internal/clients/clients.go

```diff
@@ -107,8 +107,33 @@ func restConfigFromAPIConfig(c *api.Config) (*rest.Config, error) {
 	return config, nil
 }
 
+// RestConfigGetter is an interface that provides a REST config.
+type RestConfigGetter interface {
+	// GetConfig returns an initialized Config
+	GetConfig() *rest.Config
+}
+
+// ClusterClient is a client that can be used to interact with a Kubernetes
+// cluster including getting its rest config.
+type ClusterClient interface {
+	client.Client
+	resource.Applicator
+	RestConfigGetter
+}
+
+// ApplicatorClientWithConfig is a ClusterClient that also has a rest config.
+type ApplicatorClientWithConfig struct {
+	resource.ClientApplicator
+	config *rest.Config
+}
```

Comment:
cosmetics/test: this is an odd package of data structures.

Reply:
Agreed, I had similar feeling but didn't feel motivated. 
Refactored it and get rid of most of those interfaces/structs.